src/python/pants/option/arg_splitter_test.py

```diff
@@ -98,140 +86,259 @@ def assert_not_positional(arg):
+      './pants compile test foo::',
+      expected_goals=['compile', 'test'],
+      expected_scope_to_flags={'': [], 'compile': [], 'test': []},
+      expected_positional_args=['foo::'],
+    )
+    self.assert_valid_split(
+      './pants compile test:test',
+      expected_goals=['compile'],
+      expected_scope_to_flags={'': [], 'compile': []},
+      expected_positional_args=['test:test'],
+    )
+    self.assert_valid_split(
+      './pants test test:test',
+      expected_goals=['test'],
+      expected_scope_to_flags={'': [], 'test': []},
+      expected_positional_args=['test:test'],
+    )
+    self.assert_valid_split(
+      './pants test ./test',
+      expected_goals=['test'],
+      expected_scope_to_flags={'': [], 'test': []},
+      expected_positional_args=['./test'],
+    )
+    self.assert_valid_split(
+      './pants test //test',
+      expected_goals=['test'],
+      expected_scope_to_flags={'': [], 'test': []},
+      expected_positional_args=['//test'],
+    )
 
+  def test_descoping_qualified_flags(self) -> None:
+    self.assert_valid_split(
+      './pants compile test --compile-java-bar --no-test-junit-baz foo/bar',
+      expected_goals=['compile', 'test'],
+      expected_scope_to_flags={
+        '': [], 'compile': [], 'compile.java': ['--bar'], 'test': [], 'test.junit': ['--no-baz']
+      },
+      expected_positional_args=['foo/bar'],
+    )
     # Qualified flags don't count as explicit goals.
-    self._split('./pants compile --test-junit-bar foo/bar',
-                ['compile'],
-                {'': [], 'compile': [], 'test.junit': ['--bar']}, ['foo/bar'])
-
-  def test_passthru_args(self):
-    self._split('./pants test foo/bar -- -t arg',
-                ['test'],
-                {'': [], 'test': []},
-                ['foo/bar'],
-                expected_passthru=['-t', 'arg'],
-                expected_passthru_owner='test')
-    self._split('./pants -farg --fff=arg compile --gg-gg=arg-arg -g test.junit --iii '
-                '--compile-java-long-flag src/java/org/pantsbuild/foo '
-                'src/java/org/pantsbuild/bar:baz '
-                '-- passthru1 passthru2',
-                ['compile', 'test'],
-                {
-                  '': ['-farg', '--fff=arg'],
-                  'compile': ['--gg-gg=arg-arg', '-g'],
-                  'compile.java': ['--long-flag'],
-                  'test.junit': ['--iii']
-                },
-                ['src/java/org/pantsbuild/foo', 'src/java/org/pantsbuild/bar:baz'],
-                expected_passthru=['passthru1', 'passthru2'],
-                expected_passthru_owner='test.junit')
-
-  def test_subsystem_flags(self):
+    self.assert_valid_split(
+      './pants compile --test-junit-bar foo/bar',
+      expected_goals=['compile'],
+      expected_scope_to_flags={'': [], 'compile': [], 'test.junit': ['--bar']},
+      expected_positional_args=['foo/bar'],
+    )
+
+  def test_passthru_args(self) -> None:
```

Comment:
This function is the same, modulo using keyword args and some formatting tweaks.